### PR TITLE
FFH-416 Add Setting for Disable Text Area

### PR DESCRIPTION
**Pull request by sonujangra27**
### What do you want to achieve?
- Add Setting for Disable Text Area

### PR Checklist

- [x] I have tested it locally and all functionalities are working fine.
- [x] I have compared it with mocks and all design elements are the same.
- [ ] I have tested it in IE Browser.

### How was the code tested?

- tested when the chat is assigned to bot
- tested when chat is assigned to human

### What new thing you came across while writing this code? 
-

### In case you fixed a bug then please describe the root cause of it? 
-

### Screenshot
- 

NOTE: Make sure you're comparing your branch with the correct base branch

### CODE REVIEW

**Comment by sonujangra27** [comment]
not sure about the name for appSetting for now - will change it if needed.
for now - disableTextArea this is being called as per the task details

**Review by prochnost04 — APPROVED** [review]

**Review by panwaranuj01 — APPROVED** [review]

**Review by Sanjeevsinghnegi — DISMISSED** [review]

**Comment by sonujangra27** [comment]
@Sanjeevsinghnegi @prochnost04 please check and let me know if any changes are required-  
have checked for assignee and group users - can not check the in main function because - in each place - object is of diff type

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 1]
```diff
@@ -405,6 +405,14 @@ var userOverride = {
         var MCK_GROUPMAXSIZE = appOptions.maxGroupSize;
         var MCK_ON_TAB_CLICKED = function (event) {
             console.log("In on_tab_clicked", event);
+            const details = event.data.groupDetails;
+            if (details) {
+                const assignee =
+                    details.metadata && details.metadata.CONVERSATION_ASSIGNEE;
+                const groupUsers = details.groupUsers;
+                details &&  assignee && groupUsers && KommunicateUI.toggleVisibilityOfTextArea(assignee, groupUsers);
```
Comment (Sanjeevsinghnegi) [inline webplugin/js/app/mck-sidebox-1.0.js:413]: `details &&` is not req here

**Review by Sanjeevsinghnegi — CHANGES_REQUESTED** [review]

File: webplugin/js/app/kommunicate-ui.js [hunk 2]
```diff
@@ -1904,4 +1897,43 @@ KommunicateUI = {
             }
         );
     },
+    toggleVisibilityOfTextArea: function (assignee,groupMembers) {
+        var isDisableTextAreaEnabled =
+        kommunicate &&
+        kommunicate._globals &&
+        kommunicate._globals.disableTextArea;
+        if(isDisableTextAreaEnabled){
+            if(assignee && groupMembers){
```
Comment (prochnost04) [inline webplugin/js/app/kommunicate-ui.js:1906]: we could have merged these?

File: webplugin/js/app/kommunicate-ui.js [hunk 3]
```diff
@@ -1904,4 +1897,43 @@ KommunicateUI = {
...
+                for (var i = 0; i < groupMembers.length; i++) {
+                    if (
+                        groupMembers[i].userId === assignee &&
```
Comment (prochnost04) [inline webplugin/js/app/kommunicate-ui.js:1909]: add check for groupMembers[i]

File: webplugin/js/app/kommunicate-ui.js [hunk 4]
```diff
@@ -1904,4 +1897,43 @@ KommunicateUI = {
...
+                        (
+                            groupMembers[i].roleType == '1' ||
+                            groupMembers[i].role == '2'
```
Comment (prochnost04) [inline webplugin/js/app/kommunicate-ui.js:1912]: avoid using magic numbers

File: webplugin/js/app/kommunicate-ui.js [hunk 5]
```diff
@@ -1904,4 +1897,43 @@ KommunicateUI = {
...
+                            )
+                    ) {
+                        kommunicateCommons.modifyClassList(
+                            {
+                                class: ['mck-box-form'],
+                            },
+                            'n-vis',
+                            'vis'
+                        );
+                        break;
```
Comment (prochnost04) [inline webplugin/js/app/kommunicate-ui.js:1922]: is break required here ?

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 6]
```diff
@@ -405,6 +405,14 @@ var userOverride = {
         var MCK_GROUPMAXSIZE = appOptions.maxGroupSize;
         var MCK_ON_TAB_CLICKED = function (event) {
             console.log("In on_tab_clicked", event);
+            const details = event.data.groupDetails;
```
Comment (prochnost04) [inline webplugin/js/app/mck-sidebox-1.0.js:408]: add checks

File: webplugin/js/app/mck-sidebox-1.0.js [hunk 7]
```diff
@@ -8589,6 +8629,7 @@ var userOverride = {
                         MCK_ON_TAB_CLICKED({
                             tabId: params.tabId,
                             isGroup: params.isGroup,
+                            data:params
```
Comment (prochnost04) [inline webplugin/js/app/mck-sidebox-1.0.js:8632]: formatting

**Review by prochnost04 — CHANGES_REQUESTED** [review]

File: webplugin/js/app/mck-sidebox-1.0.js (hunk 1 above)
Comment (sonujangra27) [reply webplugin/js/app/mck-sidebox-1.0.js]: done

File: webplugin/js/app/kommunicate-ui.js (hunk 2 above)
Comment (sonujangra27) [reply webplugin/js/app/kommunicate-ui.js]: done

File: webplugin/js/app/kommunicate-ui.js (hunk 3 above)
Comment (sonujangra27) [reply webplugin/js/app/kommunicate-ui.js]: done

File: webplugin/js/app/kommunicate-ui.js (hunk 4 above)
Comment (sonujangra27) [reply webplugin/js/app/kommunicate-ui.js]: used the constants

File: webplugin/js/app/kommunicate-ui.js (hunk 5 above)
Comment (sonujangra27) [reply webplugin/js/app/kommunicate-ui.js]: yeah when the widget was loading there comes a case - when it does not go into the if 
and makes the textbox visible - 
can remove it if needed - 
but it won't harm

File: webplugin/js/app/mck-sidebox-1.0.js (hunk 6 above)
Comment (sonujangra27) [reply webplugin/js/app/mck-sidebox-1.0.js]: done

File: webplugin/js/app/mck-sidebox-1.0.js (hunk 7 above)
Comment (sonujangra27) [reply webplugin/js/app/mck-sidebox-1.0.js]: done

**Review by Sanjeevsinghnegi — APPROVED** [review]

**Review by prochnost04 — APPROVED** [review]

**Pull request merged**